core/tests/ts-integration/tests/custom-account.test.ts

```diff
@@ -95,6 +112,68 @@ describe('Tests for the custom account behavior', () => {
         ).toBeAccepted([erc20BalanceChange, feeCheck]);
     });
 
+    test('Should fail transaction validation due to timestamp assertion in the validation tracer - close to the range end', async () => {
+        const now = Math.floor(Date.now() / 1000);
+        const minTimeTillEnd = testMaster.environment().timestampAsserterMinTimeTillEndSec;
+        const rangeStart = now - 10;
+        const rangeEnd = now + minTimeTillEnd / 2;
+
+        const customAccount = await deployAndFundCustomAccount(
+            alice,
+            erc20Address,
+            timestampAsserterAddress,
+            rangeStart,
+            rangeEnd
+        );
+
+        const tx = await erc20.transfer.populateTransaction(alice.address, TRANSFER_AMOUNT);
+
+        await expect(
+            sendCustomAccountTransaction(
+                tx as zksync.types.Transaction,
+                alice.provider,
+                await customAccount.getAddress(),
+                testMaster.environment().l2ChainId
+            )
+        ).toBeRejected(
+            'failed to validate the transaction. reason: Violated validation rules: block.timestamp is too close to the range end'
```

Comment:
Nit: Would match via a regex work here (like `/block\.timestamp/`)? It's usually a bad idea to tie tests to human-readable error messages.

Reply:
Agreed - this is rather a pattern that is used across all integration tests. Ideally i'd match against a defined set of error codes.